### PR TITLE
add lintr step to PR workflow

### PR DESCRIPTION
**Pull request by bernt-matthias**
also integrate latest changes (from https://github.com/galaxyproject/tools-devteam/pull/560) from devteam repo.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [x] - This PR does something else (explain below)

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -172,13 +172,50 @@ jobs:
...
+        path: ../workflow_artifacts/
+    - uses: goanpeca/setup-miniconda@v1
+      with:
+        auto-update-conda: true
+        python-version: ${{ matrix.python-version }}
+        channels: conda-forge
+        use-only-tar-bz2: true
+    - name: Install lintr
+      shell: bash -l {0}
+      run: conda install r-lintr
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:201]: Possible alternative: https://github.com/r-lib/actions/tree/master/examples

**Comment by bernt-matthias** [comment]
Seems to work -- also the caching runs if the lint step is successful. I think the last question is if the caching of the R packages is fine as it is. Maybe it should be versioned?

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -172,13 +168,63 @@ jobs:
...
+    - uses: actions/download-artifact@v2
+      with:
+        name: Workflow artifacts
+        path: ../workflow_artifacts/
+    - uses: r-lib/actions/setup-r@master
+    - name: Cache R packages
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.R_LIBS_USER }}
+        key: lintr-rpackages
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:190]:
```suggestion
        key: r_cache_${{ matrix.r-version }}
```

**Comment by bernt-matthias** [comment]
OK. Cool. Not sure about R package management. What happens if a new version of lintr or one of its requirements is available?

**Comment by nsoranzo** [comment]
> OK. Cool. Not sure about R package management. What happens if a new version of lintr or one of its requirements is available?

As it is now, the present version of lintr will be used from the cache until we change the R version in the matrix. I don't think it is a big deal, we can change that later if needed.

**Pull request merged**